org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WDateEditor.java

```diff
@@ -175,7 +175,7 @@ public void setValue(Object value)
     		Timestamp currentValue = oldValue;
     		oldValue = null;
     		getComponent().setValue(null);
-    		if (currentValue != null)
+    		if (currentValue != null && !isGridFieldVirtualUIColumn())
```

Comment:
Don't think the new method is necessary. Should work by just checking whether the field is editable.

Reply:
@hengsin Changed check for read only.